packages/boot/test/bootstrapTests/orchestration.test.ts

```diff
@@ -231,3 +231,41 @@ test.serial('stakeAtom - smart wallet', async t => {
     'delegate fails with invalid validator',
   );
 });
+
+// XXX rely on .serial to be in sequence, and keep this one last
+test.serial('revise chain info', async t => {
+  const {
+    buildProposal,
+    evalProposal,
+    runUtils: { EV },
+  } = t.context;
+
+  const agoricNames = await EV.vat('bootstrap').consumeItem('agoricNames');
+
+  const agoricBefore = await EV(agoricNames).lookup('chain', 'agoric');
+  t.like(agoricBefore, { chainId: 'agoric-3' });
+
+  await t.throwsAsync(EV(agoricNames).lookup('chain', 'hot'), {
+    message: '"nameKey" not found: "hot"',
+  });
+
+  // Revise chain info in agoricNames with the fixture in this script
+  await evalProposal(
+    buildProposal('@agoric/builders/scripts/testing/tweak-chain-info.js'),
+  );
+
+  const hotAfter = await EV(agoricNames).lookup('chain', 'hot');
+  t.deepEqual(hotAfter, { allegedName: 'Hot New Chain', chainId: 'hot-1' });
+
+  const agoricAfter = await EV(agoricNames).lookup('chain', 'agoric');
+  t.like(agoricAfter, { chainId: 'agoric-4' });
```

Comment:
Do we have requirements to handle changing a chainId change for a well-established chain? That's the sort of "earthquake" event that I think the market doesn't really contemplate. It would mean something like abandoning all the tokens that have ever gone over IBC to or from agoric-3. I'd rather not have a test that suggests that we handle this sort of thing. I'd much rather stick to "append only" changes to chain info (_for a somewhat hand-wavy notion of append_).

Reply:
I think we have a need to revise chain_id for Starship because it will have different ids for the names.

Since Starship is an E2E test, I'm fine relying on that actual test for whether the chain ids update as needed and taking this out of the test here that could be read as a product spec. I'll remove it.

Reply:
Ah. Yes, I look forward to supporting a chainId other than agoric-3 in the starship context. I'd expect it to go into agoricNames once as `agoric-4` in the first place, though.